solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -1004,24 +1002,9 @@ public void load() {
       backgroundCloser.start();
 
     } finally {
-      if (asyncSolrCoreLoad && futures != null) {
-
+      if (asyncSolrCoreLoad) {
         coreContainerWorkExecutor.submit(
-            () -> {
-              try {
-                for (Future<SolrCore> future : futures) {
-                  try {
-                    future.get();
-                  } catch (InterruptedException e) {
-                    Thread.currentThread().interrupt();
-                  } catch (ExecutionException e) {
-                    log.error("Error waiting for SolrCore to be loaded on startup", e);
```

Comment:
At least the old code was just logging on failure. I guess this was ok to not throw anything?

Reply:
yes, hence no need to throw an exception at a lower level; in task body